### ISSUE TITLE
model.fit()  is loading entire data even after providing batch_size

### ISSUE THREAD

**Issue by Abir-lal-Roy** [tensorflow/tensorflow#58120]
<details><summary>Click to expand!</summary> 
 
 ### Issue Type

Support

### Source

source

### Tensorflow Version

2.4.1

### Custom Code

No

### Current Behaviour?

```shell
I have two numpy arrays ([[512 dim vector],[512 dim vector],..,[512 dim vector]] - array looks like this) of shape (55M * 512) which are being used as features in my model.I am not able to train the model because of memory error.I put batch_size = 1024.But model.fit() is loading the entire data to the memory.
```

### Standalone code to reproduce the issue

```shell
def create_model():

    all_inputs = []
    output = []   
    text_input_1 = layers.Input(shape=(512,),name='text_1')
    
    mul = 0.56
    num_neuron = int(512*mul)
    text_output_1=layers.Dense(num_neuron, input_dim=(512), activation='relu')(text_input_1)
    all_inputs.append(text_input_1)
    
    text_input_2 = layers.Input(shape=(512,),name='text_2')
    text_output_2=layers.Dense(128, input_dim=(512), activation='relu')(text_input_2)    
    all_inputs.append(text_input_2)
    
    embedding_power= 0.49

    for c in cat_feat:
        num_unique_values = int(df_train[c].nunique())
        embed_dim = int(num_unique_values**embedding_power)#int(min(np.ceil((num_unique_values)/2), 50))
        inp = layers.Input(shape=(1,))
        out = layers.Embedding(num_unique_values + 1, embed_dim, name=c)(inp)
        out = layers.SpatialDropout1D(0.3)(out)
        out = layers.Reshape(target_shape=(embed_dim, ))(out)
        all_inputs.append(inp)
        output.append(out)    

    conc = layers.Concatenate()(output)
    x = layers.concatenate([conc, text_output_1,text_output_2],name='concat_of_text_and_cat_feat')
    
    
    # layer 0
    l = x.shape[1]
    num_hidden = int(0.553*l)
    x = layers.Dense(num_hidden,kernel_initializer='he_uniform',activation='relu')(x)
    x = layers.BatchNormalization()(x)
    
    # layer 1

    l = x.shape[1]
    num_hidden = int(0.483*l)
    x = layers.Dense(num_hidden,kernel_initializer='he_uniform',activation='relu')(x)
    x = layers.Dropout(0.15)(x)
    x = layers.BatchNormalization()(x)
    
    # layer 2

    l = x.shape[1]
    num_hidden = int(0.590*l)
    x = layers.Dense(num_hidden,kernel_initializer='he_uniform',activation='relu')(x)
    x = layers.Dropout(0.3)(x)
    x = layers.BatchNormalization()(x)
    
    preds = layers.Dense(1, activation="sigmoid")(x)
    model = Model(inputs=all_inputs, outputs=preds)

    # compile our model with a sampled learning rate.
    lr = 0.00012
    
    model.compile(
        loss="binary_crossentropy",
        optimizer=tf.keras.optimizers.Adam(learning_rate=lr),
        metrics=["AUC"],run_eagerly=True
    )  
    return model

model = create_model()

model.fit([text_train_1,text_train_2,[df_train.loc[:,f].values for f in cat_feat_nn]],
          df_train["target"].values,epochs=60,verbose=1,batch_size=1024)
#The last step is giving me error

How would I provide chunk of data (batch size) to the model for training instead of loading entire data to the memory ?
```

### Relevant log output

_No response_</details>

**Comment by tilakrayal** [comment]
@Abir-lal-Roy,
I was facing a different error while executing the given code. Kindly find the gist of it [here](https://colab.research.google.com/gist/tilakrayal/def2e49d7e72c2a416a66f20b4470afd/untitled692.ipynb) and share all dependencies to replicate the issue .

Also tensorflow version 2.4 is not actively supported. Hence, kindly update to the latest stable version 2.9 and let us know if you are facing the same issue. Thank you!

**Comment by Abir-lal-Roy** [comment]
@tilakrayal Hi, the error you are facing is because of not importing the required libraries.Let me give you all the required libraries you need to import to reproduce the issue

import tensorflow as tf
import tensorflow_hub as hub

from tensorflow.keras import layers
from tensorflow.keras import optimizers
from tensorflow.keras.models import Model, load_model
from tensorflow.keras import callbacks
from tensorflow.keras import backend as K
from tensorflow.keras import utils

from tensorflow.keras.models import Sequential
from tensorflow.keras.layers import Dense, Dropout, BatchNormalization

Once I run model.fit(), memory uses starts increasing significantly until kernal gets restarted. Because it loads these two numpy arrays (text_train_1,text_train_2) to ram.

Also I will upgrade my tf version to 2.9 as suggested.Thanks!

**Comment by tilakrayal** [comment]
@Abir-lal-Roy,
I was facing a different error while executing the code. Kindly find the gist of it [here](https://colab.research.google.com/gist/tilakrayal/513285b35c34db1710823185fcbfe709/untitled713.ipynb) and provide the required dependencies. Thank you!
